test/bootstrap_rich_text_area_test.rb

```diff
@@ -8,18 +9,39 @@ class BootstrapRichTextAreaTest < ActionView::TestCase
     setup :setup_test_fixture
 
     test "rich text areas are wrapped correctly" do
-      expected = <<~HTML
-        <div class="mb-3">
-          <label class="form-label" for="user_life_story">Life story</label>
-          <input type="hidden" name="user[life_story]" id="user_life_story_trix_input_user"/>
+      with_stub_token do
+        expected = <<~HTML
+          <div class="mb-3">
+            <label class="form-label" for="user_life_story">Life story</label>
+            <input #{'autocomplete="off"' if ::Rails::VERSION::STRING >= '7'}
+              type="hidden" name="user[life_story]" id="user_life_story_trix_input_user"/>
+            #{trix_editor_html}
+          </div>
+        HTML
+        assert_equivalent_xml expected, form_with_builder.rich_text_area(:life_story)
+      end
+    end
+
+    def trix_editor_html
+      if ::Rails::VERSION::STRING >= "7"
+        <<~HTML
+          <trix-editor class="trix-content form-control" data-blob-url-template="http://test.host/rails/active_storage/blobs/redirect/:signed_id/:filename" data-direct-upload-attachment-name="ActionText::RichText#embeds" data-direct-upload-token="token" data-direct-upload-url="http://test.host/rails/active_storage/direct_uploads" id="user_life_story" input="user_life_story_trix_input_user"/>
+        HTML
+      else
+        <<~HTML
           <trix-editor id="user_life_story" data-blob-url-template="#{data_blob_url_template}" data-direct-upload-url="http://test.host/rails/active_storage/direct_uploads" input="user_life_story_trix_input_user" class="trix-content form-control" />
-        </div>
-      HTML
-      assert_equivalent_xml expected, form_with_builder.rich_text_area(:life_story)
+        HTML
+      end
     end
 
     def data_blob_url_template
       "http://test.host/rails/active_storage/blobs/#{'redirect/' if ::Rails::VERSION::STRING >= '6.1'}:signed_id/:filename"
     end
+
+    def with_stub_token(&block)
+      return if ::Rails::VERSION::STRING < "7"
```

Comment:
I think this means you skip running the assertion, which I don't think you intended. Maybe you want to just `yield` if the Rails version is less than 7?

Reply:
You're right, thanks for catching this! `return yield` would fix it, but looking at the test as a whole again all the checks for Rails 7 make it less readable. So I now choose to have one bit of mark-up and test for >= Rails 7 and one for older versions. Makes it a bit less DRY, but I think this will make it easier to understand in the future.